pkg/controller/datavolume-controller.go

```diff
@@ -2823,14 +2823,27 @@ func (r *DatavolumeReconciler) detachPvcDeleteDv(pvc *corev1.PersistentVolumeCla
 }
 
 func updatePvcOwnerRefs(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) {
-	refs := pvc.OwnerReferences
-	for i, r := range refs {
-		if r.UID == dv.UID {
-			pvc.OwnerReferences = append(refs[:i], refs[i+1:]...)
+	var refs []metav1.OwnerReference
+	for _, pvcOwnerRef := range pvc.OwnerReferences {
+		if pvcOwnerRef.UID != dv.UID {
+			refs = append(refs, pvcOwnerRef)
 			break
 		}
 	}
-	pvc.OwnerReferences = append(pvc.OwnerReferences, dv.OwnerReferences...)
+	for _, dvOwnerRef := range dv.OwnerReferences {
+		refExists := false
+		for _, pvcOwnerRef := range pvc.OwnerReferences {
+			if dvOwnerRef.UID == pvcOwnerRef.UID {
+				refExists = true
+				break
+			}
+		}
```

Comment:
maybe extratc this as 
`findRef(pvc.OwnerReferences, dvOwnerRef.UID)` 
or 
`existsRef(dvOwnerRef.UID, pvc.OwnerReferences)`

Reply:
It's already a compact function, so extracting this loop to another func seems unnecessary.

Reply:
but it will be obvious what it does without reading two levels of for loop and break statement;
```
for _, dvOwnerRef := range dv.OwnerReferences {
		if !refExists(dvOwnerRef.UID, pvc.OwnerReferences) {
           			dvOwnerRef.BlockOwnerDeletion = nil
			        refs = append(refs, dvOwnerRef)
        }
}
```

Reply:
btw: this is square complexity, but I suppose we should never see more that a few references so it is ok.

Reply:
Right, in the OpenShift case it's actually 1x1 :joy:
Anyway, I'll extract it as I really want to make you happy today, bro.